File: Unit2-01/Typescript/stack.ts
/** 
 * This program defines a stack
 * By:      Johanna Liu
 * Version: 1.0
 * Since:   2024-03-03
 */

class Stack {
  //create stack as list
  stackList: string[]

  constructor() {
    this.stackList = []
  }

  push(pushedString: string): void {
    // push the value onto the stack
    this.stackList.push(pushedString)
  }

  showStack(): string {
    // return the stack
    let stackValues: string = ""

    for (const value of this.stackList) {
      stackValues += value + ", "
    }
    stackValues = stackValues.slice(0, -2) // Remove the trailing comma

    return stackValues
  }
}

// Export the class
export default Stack;

File: Unit2-03/Java/Stack.java
hi/*
 * This program uses stacks
 *
 * @author  Johanna Liu
 * @version 1.0
 * @since   2024-03-29
 */

import java.util.ArrayList;
import java.util.List;

/**
* This is my stack.
*/
public class Stack {
    /**
     * Constructor.
     *
     */
    private List<String> stackList = new ArrayList<String>();

    /**
    * Push a string to end of stack.
    *
    * @param item push item
    */
    public void push(String item) {
        stackList.add(item);
    }

    /**
    * Show all items on stack.
    *
    * @return the full stack
    */
    public String showStack() {
        String values = "";
        for (int i = 0; i < stackList.size(); i++) {
            values = values + stackList.get(i) + ", ";
        }
        values = values.substring(0, values.length() - 2);
        return values;
    }

    /**
     * Pops a string from stack and returns it.
     *
     * @return the item
     *
     */
    public String pop() {
        String removedItem = "";
        if (stackList.size() > 0) {
            removedItem = stackList.remove(stackList.size() - 1);
        } else {
            removedItem = "Stack Empty!";
        }
        return removedItem;
    }
}
